### PR TITLE
Augments/Prosthetic limbs now spark when shot instead of bleeding

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Title.

The amount of sparks are based on the damage dealt by the projectile with slight variation, with one spark representing 25 damage.

Also cleans up some code while I was there (not because melbert gave me a side eye or anything for not doing it at first hahaahahah)
## Why It's Good For The Game

Bleeding metal is weird, but I feel that having some extra feedback for getting shot was nice - it felt like something was missing when I was testing it out... so I added the sparks.
## Changelog
:cl:
fix: Projectiles no longer cause a blood graphic or blood splatters if they hit a limb that cant bleed
add: Prosthetics/Augments now spark when shot
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
epic ci fail

**Comment by nikothedude** [comment]
epic ci success

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -228,11 +228,11 @@
 	qdel(src)
 
 //to get the correct limb (if any) for the projectile hit message
-/mob/living/proc/check_limb_hit(hit_zone)
+/mob/living/proc/check_hit_limb_name(hit_zone)
```
Comment (Jacquerel) [inline code/modules/projectiles/projectile.dm:231]: Should this not be `check_hit_limb_zone`? It doesn't return the name of the limb.

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -270,66 +270,75 @@
...
+				else
+					new /obj/effect/temp_visual/dir_setting/bloodsplatter(target_turf, splatter_dir)
+				if(prob(33))
+					living_target.add_splatter_floor(target_turf)
+			else if (!isnull(hit_bodypart) && ((hit_bodypart.biological_state & BIO_METAL) || (hit_bodypart.biological_state & BIO_WIRED)))
+				var/random_damage_mult = LERP(0.85, 1.15, rand()) // SOMETIMES you can get more or less sparks
+				var/damage_dealt = ((damage / (1 - (blocked / 100))) * random_damage_mult)
+
+				var/spark_amount = round((damage_dealt / PROJECTILE_DAMAGE_PER_ROBOTIC_SPARK))
+				if (spark_amount)
```
Comment (Jacquerel) [inline code/modules/projectiles/projectile.dm:312]:
```suggestion
				if (spark_amount > 0)
```
Nice to be specific here

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile.dm (hunk 1 above)
Comment (nikothedude) [reply code/modules/projectiles/projectile.dm:231]: probably, when I get home I can rename it

File: code/modules/projectiles/projectile.dm [hunk 3]
```diff
@@ -270,66 +270,75 @@
...
+				var/splatter_dir = dir
+				if(starting)
+					splatter_dir = get_dir(starting, target_turf)
+				if(isalien(living_target))
+					new /obj/effect/temp_visual/dir_setting/bloodsplatter/xenosplatter(target_turf, splatter_dir)
+				else
+					new /obj/effect/temp_visual/dir_setting/bloodsplatter(target_turf, splatter_dir)
+				if(prob(33))
+					living_target.add_splatter_floor(target_turf)
+			else if (!isnull(hit_bodypart) && ((hit_bodypart.biological_state & BIO_METAL) || (hit_bodypart.biological_state & BIO_WIRED)))
```
Comment (Ghommie) [inline code/modules/projectiles/projectile.dm:307]:
```suggestion
			else if (!isnull(hit_bodypart) && (hit_bodypart.biological_state & (BIO_METAL|BIO_WIRED))
```

File: code/modules/projectiles/projectile.dm [hunk 4]
```diff
@@ -270,66 +270,75 @@
...
+				var/random_damage_mult = LERP(0.85, 1.15, rand()) // SOMETIMES you can get more or less sparks
```
Comment (Ghommie) [inline code/modules/projectiles/projectile.dm:308]:
```suggestion
				var/random_damage_mult = RANDOM_DECIMAL(0.85, 1.15) // SOMETIMES you can get more or less sparks
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile.dm (hunk 4 above)
Comment (nikothedude) [reply code/modules/projectiles/projectile.dm:308]: why did noone tell me this macro existed *sniff

**Pull request merged**